### PR TITLE
Add quantization tool and its unittest with s8s8 support

### PR DESCRIPTION
**Pull request by zhanghuanrong**
Add quantization tool with s8s8 support
      * Add unittest for existing s8s8 support operators
      * Comment ready unittest for upcomming s8s8 operator (ConvInteger, and Resize)
      * Minor change on quantization tools

### CODE REVIEW

File: onnxruntime/python/tools/quantization/onnx_quantizer.py [hunk 1]
```diff
@@ -532,6 +532,7 @@ def _get_quantize_input_nodes(self, node, input_index, qType, given_scale_name=N
                 return None
             # dynamic mode
             # Scale and Zero Points not available for this input. Add nodes to dynamically compute it
+            # TODO: Support DynamicQuantizeLinear for s8 after onnxruntime have it.
```
Comment (yufenglee) [inline onnxruntime/python/tools/quantization/onnx_quantizer.py:535]: >DynamicQuantizeLinear [](http://example.com/codeflow?start=28&length=21)

DynamicQuantizeLinear only supports uint8

**Comment by yufenglee** [comment]
it is assumed that the range of int8 is [-127, 127]. This is not true anymore for activation. So we need to differentiate if it is weight or activation. for weight, it is [-127, 127], for activation it should be [-128, 127].

[https://github.com/microsoft/onnxruntime/blob/3c79f3055f4b9d3a67c7ee814f8ebc9cba1bda12/onnxruntime/python/tools/quantization/quant_utils.py#L195](https://github.com/microsoft/onnxruntime/blob/3c79f3055f4b9d3a67c7ee814f8ebc9cba1bda12/onnxruntime/python/tools/quantization/quant_utils.py#L195)
[https://github.com/microsoft/onnxruntime/blob/3c79f3055f4b9d3a67c7ee814f8ebc9cba1bda12/onnxruntime/python/tools/quantization/quant_utils.py#L105](https://github.com/microsoft/onnxruntime/blob/3c79f3055f4b9d3a67c7ee814f8ebc9cba1bda12/onnxruntime/python/tools/quantization/quant_utils.py#L105)

**Comment by yufenglee** [comment]
And could you please add assertion to fail s8u8 quickly.

---
In reply to: [991186598](https://github.com/microsoft/onnxruntime/pull/10007#issuecomment-991186598) [](http://example.com/codeflow?ancestors=991186598)

File: onnxruntime/python/tools/quantization/onnx_quantizer.py (hunk 1 above)
Comment (zhanghuanrong) [reply onnxruntime/python/tools/quantization/onnx_quantizer.py:535]: Yeah, currently it is, and it is hard to decide type from output type?
But It is finally will be asked if S8S8 dynamic is used, we may need other solutions.

**Comment by zhanghuanrong** [comment]
test need x64 s8s8 support first.

**Review by yufenglee — APPROVED** [review]
:shipit:

**Pull request merged**